src/net/win32/wif.c

```diff
@@ -46,14 +47,24 @@ static int if_list_gaa(net_ifaddr_h *ifh, void *arg)
 		goto out;
 	}
 
-	ret = (*u.gaa)(AF_UNSPEC, flags, NULL, addrv, &len);
+	/* try with stack-allocated addrv at first */
+	paddrv = addrv;
+	addrv_allocated = false;
+retry:
```

Comment:
we should avoid a retry loop here, since there is a chance that the code will be stuck
in the loop forever.